### PR TITLE
Feature/conditionally required fields

### PR DESCRIPTION
**Pull request by ValJed**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [ ] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/schema/ui/apos/mixins/AposInputFollowingMixin.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ export default {
         parentFollowing[`<${key}`] = val;
       }
 
-      for (const field of this.field.schema) {
+      for (const field of this.schema) {
```
Comment (boutell) [inline modules/@apostrophecms/schema/ui/apos/mixins/AposInputFollowingMixin.js:17]: Have we confirmed this mixin is only used in contexts where the schema computed property exists?

Comment (boutell) [reply modules/@apostrophecms/schema/ui/apos/mixins/AposInputFollowingMixin.js:17]: ... I confirmed that. Looks good.

**Review by boutell — DISMISSED** [review]

File: modules/@apostrophecms/schema/index.js [hunk 2]
```diff
@@ -484,6 +484,82 @@ module.exports = {
...
+              return false;
+            };
+
+            // Stop there, this is an external condition thus
+            // does not need to be checked against doc fields.
+            continue;
+          }
+
+          if (val.min) {
+            if (destinationKey < val.min) {
```
Comment (haroun) [inline modules/@apostrophecms/schema/index.js:530]: we should flatten this `if`

**Review by haroun — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/schema/ui/apos/lib/conditionalFields.js [hunk 3]
```diff
@@ -103,79 +110,77 @@ export function isExternalCondition(conditionKey) {
...
       const fieldValue = values[key];
 
       if (Array.isArray(fieldValue)) {
-        result = fieldValue.includes(val);
-        break;
+        return fieldValue.includes(val);
+      }
+
+      if (val.min) {
+        return fieldValue >= val.min;
```
[the remark this answers is not in the document]
Comment (falkodev) [reply modules/@apostrophecms/schema/ui/apos/lib/conditionalFields.js:174]: Done

File: modules/@apostrophecms/schema/index.js (hunk 2 above)
Comment (falkodev) [reply modules/@apostrophecms/schema/index.js:530]: Done

**Review by haroun — APPROVED** [review]

**Pull request merged**